src/full/Agda/Interaction/BasicOps.hs

```diff
@@ -117,57 +117,56 @@ giveExpr force mii mi e = do
     mv <- lookupMeta mi
     -- In the context (incl. signature) of the meta variable,
     -- type check expression and assign meta
-    withMetaInfo (getMetaInfo mv) $ do
```

Comment:
Since you remove `withMetaInfo` here, could you please state as a precondition for `giveExpr` that it expects to be called in the context of the meta variable.